catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -59,55 +60,72 @@
 
     private int subDirLevels;
 
+    private PeriodicBatchExecutor<Metacard> periodicDrain;
+
     private boolean enableBackupPlugin = true;
 
     public CatalogBackupPlugin() {
+
+        init();
+    }
+
+    private void init() {
         subDirLevels = 0;
+        setPeriodicDrain(new PeriodicBatchExecutor<>(1000, 60, TimeUnit.SECONDS));
+        getPeriodicDrain().execute(this::backup);
```

Comment:
The setters should only validate the value they are given and save them. The `init()` method should contain the logic and be called by blueprint.